WalletWasabi.Fluent/ViewModels/Wallets/Send/LabelSelectionViewModel.cs

```diff
@@ -32,18 +32,30 @@ public LabelSelectionViewModel(Money targetAmount)
 
 	public IEnumerable<LabelViewModel> LabelsBlackList => AllLabelsViewModel.Where(x => x.IsBlackListed);
 
-	public Pocket[] AutoSelectPockets()
+	public Pocket[] AutoSelectPockets(SmartLabel recipient)
 	{
 		var knownPockets = NonPrivatePockets.Where(x => x.Labels != CoinPocketHelper.UnlabelledFundsText).ToArray();
 		var unknownPockets = NonPrivatePockets.Except(knownPockets).ToArray();
 		var privateAndUnknownPockets = _allPockets.Except(knownPockets).ToArray();
 		var privateAndKnownPockets = _allPockets.Except(unknownPockets).ToArray();
+		var knownByRecipientPockets = knownPockets.Where(x => x.Labels.Any(label => recipient.Labels.Contains(label))).ToArray();
+		var onlyKnownByRecipientPockets = knownByRecipientPockets.Where(x => x.Labels.All(label => recipient.Labels.Contains(label))).ToArray();
 
 		if (_privatePocket.Amount >= _targetAmount)
 		{
 			return new[] { _privatePocket };
 		}
 
+		if (onlyKnownByRecipientPockets.Sum(x => x.Amount) >= _targetAmount)
```

Comment:
IMO this should be before selecting the private pocket if possible. We should get rid of non private coins whenever possible if that does not hurt our privacy.

If you use `onlyKnownByRecipientPockets` the recipient would learn nothing new about your wallet and the privacy level of the wallet would increase after that tx.

But by using private coins, the recipient would know more about our wallet after making this transaction (the coins that they already knew about + the change).

Reply:
I asked @molnard about what should be the best to do. 
And you are right in the case if the Labels are consistent and work as expected.
To not rely on this we agreed to go with the atom safe solution. Use the private funds first.

Reply:
But what is the point of the labels if we don't rely on them?

I guess everywhere in the software we rely on a correct usage by the user.

Reply:
@molnard Can we rediscuss this? I think @yahiheb has a point. We very much rely on the labels even on the Privacy Control screen.

Reply:
Ok, I will move this case to the first place.